### ISSUE TITLE
We need a “Mark as having received RF commitment” button

### ISSUE THREAD

**Issue by sideshowbarker** [w3c/ash-nazg#108]
At, e.g., https://labs.w3.org/hatchery/repo-manager/pr/id/w3c/webdriver/1120 we need a button that says something like **Mark as having received RF commitment** — optionally with a way to indicate a URL for some record/documentation (e.g., issue comment) of the contributor having made the actual RF commitment.

For that specific PR, https://github.com/w3c/webdriver/pull/1120, I asked the PR submitter for a statement agreeing to an RF commitment, and at https://github.com/w3c/webdriver/pull/1120#issuecomment-334952460, they’ve already responded to provide it.

But right now at https://labs.w3.org/hatchery/repo-manager/pr/id/w3c/webdriver/1120 I have no button to push to indicate we got that commitment.

Using the **Mark as non-substantive** button wouldn’t be appropriate, because the change is in fact a substantive change.

Also, the contributor is not a member of the associated working group for the spec, and doesn’t have a W3C account — and shouldn’t need to have a W3C account just to contribute a spec change. Instead the system should have a way to record that we’ve received their commitment.

Lacking that, the only way we can merge the change from that PR branch to the master branch is for a repo admin to ignore/override the IPR check.

**Comment by dontcallmedom** [comment]
@sideshowbarker while I'd be happy to have such a mechanism, my understanding is that getting an RF commitment right now requires going through the non-member licensing commitment process (e.g. [for WebDriver](https://www.w3.org/2004/01/pp-impl/49799/nmlc)).

I actually had to go through this for a contribution to a spec in the WebRTC WG, and it currently involves snail-mail (!)…

Could you start a thread in team-legal to see if we can use a lower-bar process for this kind of contributions?

**Comment by sideshowbarker** [comment]
> @sideshowbarker while I'd be happy to have such a mechanism, my understanding is that getting an RF commitment right now requires going through the non-member licensing commitment process (e.g. for WebDriver).

Thanks for the pointer. But I notice that the entire page at https://www.w3.org/2004/01/pp-impl/49799/nmlc talks about “organizations” — “Organizations that are not W3C Members may make licensing commitments for the deliverables of a W3C Working Group… To do so, the organization should have an individual authorized to make such licensing commitments submit a signed statement to W3C, **on the organization's letterhead**.“

Pretty clearly that’s a complete mismatch for the way we these days we commonly get contributions by way of GitHub PRs from individual unaffiliated contributors — who even if they work for some particular company are making their contributions on their own time as individuals and not on behalf of their companies.

> Could you start a thread in team-legal to see if we can use a lower-bar process for this kind of contributions?

**Comment by dontcallmedom** [comment]
I would distinguish "unaffiliated contributors" from "affiliated contributors doing it on their own time", since the legal assessment for the latter is probably different from the former.

In any case, I'm very much in agreement we need better and simpler processes for these types of contributions, and will be happy to reflect them in the tool once they've been designed. Will you start the email thread, or do you prefer me to?

**Comment by sideshowbarker** [comment]
> Will you start the email thread, or do you prefer me to?

I’d prefer if you could

**Comment by wseltzer** [comment]
@sideshowbarker you say above "and shouldn’t need to have a W3C account just to contribute a spec change." Shouldn't we be able to make it easy enough to get a public W3C account that we can ask them to link a W3C account with github for recording commitments?

**Comment by sideshowbarker** [comment]
> Shouldn't we be able to make it easy enough to get a public W3C account that we can ask them to link a W3C account with github for recording commitments?

No, I think we shouldn’t force contributors to create a new account for which many or most of them basically have no use.

We want to be encouraging people to contribute to the work, don’t we? We want to be welcoming their technical contributions and thank them — not, by making them jump through an unnecessary additional hoop, discourage them and give them a disincentive to contribute.

We all know ourselves already about the user-unfriendliness of account proliferation and the feeling of account fatigue it creates — how much we personally dislike sites that require us to create accounts we don’t really need. Let’s please not be those people. Let’s not do that to others.

Anybody who submits a GitHub PR already has a GitHub account, and that rightly should be the only account they need in order to get a GitHub PR merged.

Whatever additional workflow GitHub PR contributors need to follow in order to make an RF commitment, they should ideally just be able to do it within GitHub — or at least in some way that tightly integrates with GitHub without requiring them to create a new account somewhere else.

**Comment by dontcallmedom** [comment]
I agree in principle with Mike about the cost of requiring an account, but I'm also keenly aware of the risks of solely relying on existing github accounts for tracking IPR contributions (e.g. what happens if/when github goes away or changes so much we can't keep using it).

It may be that this can be made less problematic with proper archival. But I think that @wseltzer 's approach where we significantly reduce the cost of creating a W3C account (e.g. by building it from a github account, possibly delegating by default authorization to github for these accounts) should be on the table as well.

I guess we need to map out the legal requirements, the UX goals and the technical constraints of our existing platforms.

**Comment by dontcallmedom** [comment]
linking this to the [related W3C CG process issue](https://github.com/w3c/w3process/issues/67)

**Comment by wseltzer** [comment]
We need to strike the right balance between ease-of-use and strength of patent commitments, so that implementers and users of the spec have confidence they're getting a good spec with W3C's RF patent assurance. We should make the workflow as simple as possible while capturing the information we and relying parties need. 

How can we make it super-simple for someone to create a linked W3C account, giving them the notice that they are interacting with and making commitments to W3C?

**Comment by rigow** [comment]
@wseltzer my main issue is non-repudiation. So far, buttons were discussed. Fair enough. But if we talk about a potential fantastillion $$ at stake, everybody will deny the commitment unless there is a rock-solid proof that the commitment was given. Does the imagined button do that?

**Comment by wseltzer** [comment]
@rigow Yes, large and reliance-worthy transactions happen by keypress all the time.

**Comment by rigow** [comment]
and it is easy to kill them as they are no proof. Anybody can alter or push buttons for me. Digital documents have no age or signature. Ok, I agree that courts are too stupid to understand that so far. But that doesn't make an argumentation based on their stupidity more reliable or future proof. Remember, shrink-wrap only works in the US. And even there people start to question their validity. We will be there soon when using OK buttons for multimillion dollar transactions. 
On the other hand, W3C holds no patents and makes no products, thus has no risk and can do whatever in this space. The victims will be companies relying on the Specs and that they are not encumbered. So every button or even a notice will do. (By contributing you agree to.. https://www.xkcd.com/501/ )
